### PR TITLE
Angular testing

### PR DESCRIPTION
**Pull request by DanielDavidyan**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    path: 'products',
+    loadChildren: () => import('./products/product.module').then(m => m.ProductsModule)
+  },
+  {
+    path: 'cart',
+    loadChildren: () => import('./cart/cart.module').then(m => m.CartModule)
+  }
+];
+
+// const routes: Routes = [
```
Comment (1Like1Prayer) [inline src/app/app-routing.module.ts:15]: remove

File: src/app/products/product-service/product.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import {BehaviorSubject} from 'rxjs';
+import {Product} from '../../models/stock.model';
+import {HttpClient} from '@angular/common/http';
+
+describe('ProductService', () => {
+  let service: ProductsService;
+  const productName = 'milk';
+  const productLimit = 15;
+  const product: Product = {name: 'milk', description: 'fresh', image: 'www.milk.com', limit: 10, price: 10};
+  // when(service.getProduct(productName)).thenReturn(of(product));
```
Comment (1Like1Prayer) [inline src/app/products/product-service/product.service.spec.ts:13]: remove

Comment (1Like1Prayer) [inline src/app/products/product-service/product.service.spec.ts:13]: remove

File: src/app/products/product-service/product.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      ],
+      providers: [ProductsService, HttpClient],
+    });
+  }));
+  beforeEach(() => {
+    service = TestBed.inject(ProductsService);
+    service.products = new BehaviorSubject<Product[]>([product]);
+  });
+
+  test('should be created', () => {
```
Comment (1Like1Prayer) [inline src/app/products/product-service/product.service.spec.ts:28]: test --> it

File: src/app/products/product-service/product.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    expect(service).toBeTruthy();
+  });
+
+  it('getProducts', () => {
```
Comment (1Like1Prayer) [inline src/app/products/product-service/product.service.spec.ts:32]: all tests should be in the format of 'should ....'

File: src/app/products/product-service/product.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    const products = [product];
+    expect(service.getProducts()).toEqual(new BehaviorSubject(products));
+  });
+
+  it('getProduct', () => {
```
Comment (1Like1Prayer) [inline src/app/products/product-service/product.service.spec.ts:37]: check what happens if you try to get a product that doesn't exist

File: src/app/products/product-service/product.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    expect(service.getProduct(productName)['destination']['_value'][0]).toEqual((product));
+  });
+
+  it('updateLimit', () => {
```
Comment (1Like1Prayer) [inline src/app/products/product-service/product.service.spec.ts:41]: check what happens when you try to update a product that doesn't exist
check what happens when you try to update with a number that exceeds the product limit

File: src/app/products/product-service/product.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    expect(service).toBeTruthy();
+  });
+
+  it('getProducts', () => {
+    const products = [product];
+    expect(service.getProducts()).toEqual(new BehaviorSubject(products));
+  });
+
+  it('getProduct', () => {
+    expect(service.getProduct(productName)['destination']['_value'][0]).toEqual((product));
```
Comment (1Like1Prayer) [inline src/app/products/product-service/product.service.spec.ts:38]: try to remove the `['destination']['_value']`

**Review by 1Like1Prayer — CHANGES_REQUESTED** [review]

File: src/app/products/product-service/product.service.spec.ts (hunk 5 above)
Comment (DanielDavidyan) [reply src/app/products/product-service/product.service.spec.ts:37]: I added a test

File: src/app/products/product-details/product-details.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+import {ProductsService} from '../product-service/products.service';
+import {instance, mock} from 'ts-mockito';
+import {CartProductsService} from '../../cart/cart-service/cart-products.service';
+import {ActivatedRoute} from '@angular/router';
+
+describe('ProductDetailsComponent', () => {
+  let component: ProductDetailsComponent;
+  let fixture: ComponentFixture<ProductDetailsComponent>;
+  const mockProductService: ProductsService = mock(ProductsService);
+  const router: any = mock(ActivatedRoute);
```
Comment (1Like1Prayer) [inline src/app/products/product-details/product-details.component.spec.ts:13]: check if there is a type available.

File: src/app/products/product-service/product.service.spec.ts [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+  it('should get the products', () => {
+    const products = [product];
+    expect(service.getProducts()).toEqual(new BehaviorSubject(products));
+  });
+
+  it('should get an existing product', () => {
+      let productFromService = {};
+      service.getProduct(productName).subscribe(stockProduct => productFromService = stockProduct);
+      console.log('exist');
```
Comment (1Like1Prayer) [inline src/app/products/product-service/product.service.spec.ts:40]: remove console

File: src/app/products/product-service/product.service.spec.ts [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+      expect(productFromService).toEqual(product);
+    }
+  );
+
+  it('should get an non-existent product', () => {
+      let productFromService = {};
+      service.getProduct(productNameDoesntExist).subscribe(stockProduct => productFromService = stockProduct);
+      console.log('doesnt exist');
```
Comment (1Like1Prayer) [inline src/app/products/product-service/product.service.spec.ts:48]: remove console

File: src/app/products/product-service/product.service.spec.ts [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+
+  it('should get an existing product', () => {
+      let productFromService = {};
+      service.getProduct(productName).subscribe(stockProduct => productFromService = stockProduct);
+      console.log('exist');
+      expect(productFromService).toEqual(product);
+    }
+  );
+
+  it('should get an non-existent product', () => {
```
Comment (1Like1Prayer) [inline src/app/products/product-service/product.service.spec.ts:45]: `should return undefined when trying to get a non existent product`

File: src/app/products/product-service/product.service.spec.ts [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+      let productFromService = {};
+      service.getProduct(productNameDoesntExist).subscribe(stockProduct => productFromService = stockProduct);
+      console.log('doesnt exist');
+      expect(productFromService).toEqual(undefined);
+    }
+  );
+
+  it('should be updateLimit', () => {
```
Comment (1Like1Prayer) [inline src/app/products/product-service/product.service.spec.ts:53]: `should product update limit when called`

File: src/app/cart/cart-list/cart-list.component.spec.ts [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+    fixture = TestBed.createComponent(CartListComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('checkout', () => {
```
Comment (1Like1Prayer) [inline src/app/cart/cart-list/cart-list.component.spec.ts:42]: `should checkout from cart when called`

**Review by 1Like1Prayer — CHANGES_REQUESTED** [review]

**Pull request merged**